module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -509,6 +520,27 @@ public function init()
         $this->sessionCache = $factory($namespace);
     }
 
+    /**
+     * Establish INN-Reach database connection
+     *
+     * @return resource|void
+     */
+    protected function getInnReachDb()
+    {
+        if (!isset($this->innReachDb)) {
+            try {
+                $conn_string = $this->config['InnReach']['sierra_db'];
+                $connection = pg_connect($conn_string);
+                $this->innReachDb = $connection;
+            } catch (\Exception $e) {
+                $this->logWarning("INN-Reach: Could not connect to the Sierra database: {$e}");
+                $this->innReachDb = null;
+            } finally {
+                return $this->innReachDb;
```

Comment:
If you call this function more than once, it won't return anything after the first call, because it currently only returns innReachDb when it was not previously set. I think the return statement should be moved to the very bottom of the function, outside of the if statement, to guarantee that it is returned whether or not it had to be initialized on the current call.